indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,6 +161,8 @@ public IndexTask(
     super(
         // _not_ the version, just something uniqueish
         makeId(id, ingestionSchema),
+        String.format("index_%s", makeDataSource(ingestionSchema)),
```

Comment:
I think having IndexTask use the existing constructor with ```makeId(id, ingestionSchema)``` for both the id and groupId, or adding a new (id, taskResource, dataSource, interval) constructor, would both be fine.